build-tools/src/main/java/org/elasticsearch/gradle/testclusters/RunTask.java

```diff
@@ -101,6 +104,23 @@ public void setApmServerEnabled(Boolean apmServerEnabled) {
         this.apmServerEnabled = apmServerEnabled;
     }
 
+    @Option(option = "with-plugins", description = "Run distribution with plugins installed")
+    public void setPlugins(String plugins) {
+        this.plugins = Arrays.asList(plugins.split(","));
+        for (var cluster : getClusters()) {
+            for (String plugin : this.plugins) {
+                cluster.plugin(":plugins:" + plugin);
```

Comment:
plain FYI. I noticed this is incompatible with Gradle Configuration cache. that's likely not a big deal for using `./gradlew run` at the moment, but we will rework or remove cluster.plugin(String) and cluster.module(String) at one point as it breaks configuration compatibility.